### DISCUSSION TITLE
Some questions regarding L2 EVM of op-stack chain

### DISCUSSION THREAD

**Discussion by nitantchhajed** [ethereum-optimism/developers#398] [Core Protocol] [answered]
Can someone please provide me answers to these questions for op-stack chain ?

1. Does op-geth EVM support deploying existing compiled contracts with an EVM version of ‘Paris’ or later, without additional code modifications?

2.Do op-geth support the ecrecover precompile at address(1)?

3. Can op-geth support pre-EIP155 transactions?

4. Are there any hidden transaction costs outside of EVM gas usage?

5. Does op-geth work with the Foundry tool suite?

**Comment by nitantchhajed** [comment]
@sbvegan can you help me with this?

**Reply by opfocus** [reply]
@Chomtana

**Reply by Chomtana** [reply, marked as answer]
1. I think yes op-geth supports up to Shanghai (Like transient storage isn't supported yet)
2. op-geth should support all ETH mainnet precompile up to Shanghai. So it should support address(1)
3. You need to pass `--rpc.allow-unprotected-txs` flag to op-geth in order to allow pre-EIP155 transactions
4. L1 and DA/Blob cost in addition to gas fee
5. I think so, OP stack contracts are developed using foundry.

**Reply by nitantchhajed** [reply]
thanks
